interpretation/src/main/kotlin/com/hartwig/actin/treatment/TreatmentRankingModel.kt

```diff
@@ -51,57 +51,53 @@ class TreatmentRankingModel(
                     it.characteristics.tumorMutationalLoad + it.characteristics.tumorMutationalBurden
         }.filterNotNull()
 
+
         val treatmentEvidencesWithTarget = treatmentEvidencesWithTargets(actionables)
         val scoredTreatments = treatmentEvidencesWithTarget.map { evidenceWithTarget ->
             evidenceWithTarget to scoringModel.score(evidenceWithTarget.treatmentEvidence)
         }
         val scoredTreatmentsWithDuplicatesDiminished = groupEvidenceForDuplicationAndDiminishScores(scoredTreatments)
 
-        return scoredTreatmentsWithDuplicatesDiminished
-            .map { it.key.treatment to it.value }
-            .groupBy { it.first }
-            .map { TreatmentRankResult(it.key, it.value.flatMap { t -> t.second }) }
+        return scoredTreatmentsWithDuplicatesDiminished.entries.groupBy { it.key.treatment }
+            .map { TreatmentRankResult(it.key, it.value.map { g -> g.key.target }.toSet(), it.value.flatMap { g -> g.value }) }
             .sorted()
     }
 
-    private fun groupEvidenceForDuplicationAndDiminishScores(scoredTreatmentEntries: Sequence<Pair<TreatmentEvidenceWithTarget, EvidenceScore>>) =
-        scoredTreatmentEntries.groupBy {
+    private fun groupEvidenceForDuplicationAndDiminishScores(
+        scoredTreatmentEntries: Sequence<Pair<TreatmentEvidenceWithTarget, EvidenceScore>>
+    ): Map<DuplicateEvidenceGrouping, List<EvidenceScore>> {
+        val grouped = scoredTreatmentEntries.groupBy {
             DuplicateEvidenceGrouping(
                 it.first.treatmentEvidence.treatment,
                 it.first.target,
                 it.second.scoringMatch.tumorMatch,
                 (it.second.score > 0)
             )
-        }.mapValues { entry ->
-            saturatingDiminishingReturnsScore(entry.value.map { it.second })
         }
+        val diminishedScores = grouped.mapValues { entry ->
+            val originalScores = entry.value.map { it.second }
+            val newScores = saturatingDiminishingReturnsScore(originalScores)
+            newScores
+        }
+        return diminishedScores
+    }
 
     private fun treatmentEvidencesWithTargets(actionables: Sequence<Actionable>) =
         actionables.flatMap { actionable ->
-            actionable.evidence.treatmentEvidence.asSequence().map { treatmentEvidence ->
+            val evidenceTargets = actionable.evidence.treatmentEvidence.asSequence().map { treatmentEvidence ->
```

Comment:
I don't think we need the `val evidenceTargets` you can just return it directly.